### PR TITLE
[FLYW-43] Group lab edition

### PR DESCRIPTION
**Pull request by roycollins**
Add lab edition settings for groups and projects.

Lets verify the tests are accurate with regard to the business logic around when is_lab can change and what values should be propagated throughout. 

One note is that this PR will need the modified providers branch to run successfully. 

### Review Checklist

- Tests were added to cover all code changes
- Is there a DB upgrade or fix?
  - If so is it covered by integeration tests and run against dev?
- Would this PR benefit from a test plan? If so, is one provided?
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md
- Extra care is taken and log messages are created for [Critical Data](https://github.com/flywheel-io/core/wiki/Critical-Data-Flows)

### CODE REVIEW

File: api/handlers/containerhandler.py [hunk 1]
```diff
@@ -325,6 +325,12 @@ def post(self, cont_name):
             # Validate any changes to storage providers
             providers.validate_provider_updates({}, payload.get('providers'), self.user_is_admin)
 
+            if 'is_lab' in payload and not self.user_is_admin:
+                self.abort(403, 'Only admins can specify lab_edition')
```
Comment (jody-zeitler) [inline api/handlers/containerhandler.py]: Only admins can specify is_lab

File: api/handlers/containerhandler.py [hunk 2]
```diff
@@ -358,6 +364,8 @@ def put(self, cont_name, **kwargs):
                 r_payload[key] = payload[key]
 
         if cont_name == 'projects':
+            if 'is_lab' in payload and not self.user_is_admin:
+               self.abort(403, 'Only admins can specify lab_edition')
```
Comment (jody-zeitler) [inline api/handlers/containerhandler.py]: Only admins can specify is_lab

File: api/handlers/containerhandler.py (hunk 1 above)
Comment (roycollins) [reply api/handlers/containerhandler.py]: Good call.   Fixed those!

**Comment by ehlertjd** [comment]
I think we still want to make use of the `editions` field as discussed in prior iterations of this feature: #1183 - Lab group flag and #1477 - Center flag

I think those PRs are a good place to start on getting some background for this feature.

**Comment by roycollins** [comment]
I have converted the format of the model to use the editions array.  Please let me know if this looks to be the desired approach

**Comment by roycollins** [comment]
Superseded by https://github.com/flywheel-io/core/pull/1647

**Pull request closed without merging**